esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -1970,6 +1970,36 @@ void CTpWrapper::getAttPath(const char* Path,StringBuffer& returnStr)
     JBASE64_Decode(Path, returnStr);
 }
 
+void CTpWrapper::getTpServiceTargets(double version, const char* serviceType, const char* serviceName, IArrayOf<IConstTpServiceTarget>& serviceTargets)
+{
+    Owned<IPropertyTreeIterator> services = queryComponentConfig().getElements("services");
+    ForEach(*services)
+    {
+        IPropertyTree& service = services->query();
+        const char* type = service.queryProp("@type");
+        if (isEmptyString(type) || (!isEmptyString(serviceType) && !strieq(serviceType, type)))
+            continue;
+
+        const char* name = service.queryProp("@name");
+        if (isEmptyString(name) || (!isEmptyString(serviceName) && !strieq(serviceName, name)))
+            continue;
+
+        //Only show the public services for now
+        if (!service.getPropBool("@public"))
+            continue;
+
+        Owned<IEspTpServiceTarget> serviceTarget = createTpServiceTarget();
+        serviceTarget->setName(name);
+        serviceTarget->setType(type);
+        serviceTarget->setPort(service.getPropInt("@port"));
+        if (service.hasProp("@tls")) //if not set, default to ?
```

Comment:
I would remove the comment, I think it's fine to treat absence of this property as false.
So would prob. change to:
```
   if (service.getPropBool("@tls"))
```

Reply:
Will change